File: Creational/Prototype/Java/products/Keyboard.java
package patterns.Creational.Prototype.Java.products;

public class Keyboard
{
	/**
	 * @param int
	 */
	public Keyboard(final int newKeyQty)
	{
		keyQty = newKeyQty;
	}

	/**
	 * @param Keyboard
	 */
	public Keyboard(final Keyboard keyboard)
	{
		this.keyQty = keyboard.getKeyQty();
	}

	/**
	 * @return int
	 */
	public int getKeyQty()
	{
		return keyQty;
	}

	/**
	 * @return Keyboard
	 */
	public Keyboard clone()
	{
		return new Keyboard(this);
	}

	private final int keyQty;
}

File: Creational/Prototype/PHP/products/Keyboard.php
<?php

namespace Creational\Prototype\PHP\products;

class Keyboard
{
    /**
     * @param int $newKeyQty
     */
    public function __construct(int $newKeyQty)
    {
        $this->keyQty = $newKeyQty;
    }

    /**
     * @return int
     */
    public function getKeyQty(): int
    {
        return $this->keyQty;
    }

    /**
     * @return Keyboard
     */
    public function clone(): Keyboard
    {
        return new self($this->getKeyQty());
    }

    private $keyQty;
}
